### ISSUE TITLE
2 conflicting `proj` migrations (`9.1.1` & `9.2.0`)

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/conda-forge-pinning-feedstock#4271]
Currently there are 2 `proj` migrations running that are trying to update to different `proj` versions. Should we close one of these out (presumably the older one)?

Also somewhat curious how we wound up with 2 of these (as opposed to closing the old one out)

cc @conda-forge/core @conda-forge/proj-4

**Comment by xylar** [comment]
The discussion I started on the `poppler` issue probably belongs better in this issue (see https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4270#issuecomment-1481576329).

A lot of package depend on `proj`.  A migration can take many months, long enough for one or more new versions of `proj` to be released.  Is your sense that we need to have a matrix of supported versions where we drop the oldest one only after migration has finished and add new ones as they are released?  That certainly does seem sensible to me but if we do that for many dependencies the load on CI is going to get pretty heavy.  It also doesn't seem like the way pinning is typically updated so I it wasn't clear to me that was the preferred practice.  Seemingly not to other maintainers of the scientific python stack either.

**Comment by xylar** [comment]
@ocefpaf, do you have thought on this?  Do you want to ping others that should be involved in the discussion?

**Comment by hobu** [comment]
My personal opinion is the pinning of PROJ is too constrictive in general for the geospatial packages. This was a safer thing to do when PROJ was in more flux, but while the release cadence of PROJ hasn't really slowed down, the API and behavior changes in the major versions have definitely eased up. A lot of the recent release changes are bug fixes, updated support data, performance improvements, and edge-case behavioral changes. Most of the packages could probably pin to `9.x.x` and be quite fine, but that's not to say there would not be differences in coordinate transformations or coordinate description outputs that were produced. 

@rouault may have different thoughts.

**Comment by rouault** [comment]
I'm not too sure how dependency management works without pinning, but the only constraint is that if a package (typically GDAL) is built against PROJ 9.x, it must be run against PROJ 9.y with y >= x in case a package uses new symbols, or new added enumeration values, etc. Other than that, I agree with @hobu that the differences of behaviour in the 9.x series are modest from the point of view of most PROJ using software and likely don't require a strict pinning.

**Comment by ocefpaf** [comment]
Based on what both @hobu and @rouault mentioned above seems like we could relax the pinning to x.x instead of x.x.x but the ABI tracker doesn't report 100% backwards compatibility though. See https://abi-laboratory.pro/?view=timeline&l=proj

It is quite save to have both migrations running b/c of the tight pin. It is just annoying that we are building so many variants.

**Comment by rouault** [comment]
> See https://abi-laboratory.pro/?view=timeline&l=proj

The problems it raises don't make sense to me. Issues with registers being different... ? looks like a different compiler might have been used.

**Comment by ocefpaf** [comment]
> The problems it raises don't make sense to me. Issues with registers being different... ? looks like a different compiler might have been used.

We can try x.x in conda-forge then. It will sure make out migrations easier!

**Comment by mfansler** [comment]
We can wrap up **proj911** today. We need a core member to merge [the PR on `fimex`](https://github.com/conda-forge/fimex-feedstock/pull/50).

The last one, `r-lwgeom` should be solvable once the `r-sf` builds with **proj911** are indexed. I will take care of that.

**Comment by jaimergp** [comment]
Might need a rerender, I think it's failing.

**Comment by beckermr** [comment]
FYI, I just fixed a bug in the bot-rerun labels that caused PRs to not be reissued. Any label added in the past week or so was affected. Sorry. :(

**Comment by ocefpaf** [comment]
> Might need a rerender, I think it's failing. The test data is corrupt.

Looks like upstream is no longer serving the test data at https://wiki.met.no/fimex/download. The URL is no longer responding :-/

See https://github.com/conda-forge/fimex-feedstock/pull/57

**Comment by mfansler** [comment]
All done https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4449 🎉 

I think I've caught all the ones that hit the bot-rerun issue, at least for `r-*` feedstocks.

**Comment by hmaarrfk** [comment]
closed as now we have 3 other poppler migrations that are simultaneous. will open a new issue for that.
